rust-runtime/aws-smithy-types/src/retry.rs

```diff
@@ -99,12 +99,15 @@ impl FromStr for RetryMode {
 
     fn from_str(string: &str) -> Result<Self, Self::Err> {
         let string = string.trim();
+
+        #[cfg(aws_sdk_orchestrator_mode)]
```

Comment:
This flag can only be used for tests. Customers aren't expected to set it, so it won't be set for actual runtime code.

An alternative approach to accomplish this conditional check is to just always allow adaptive to be used, but then verify in the middleware or in `make_operation` that it's not adaptive.